cmd/tusd/cli/serve.go

```diff
@@ -92,7 +113,42 @@ func Serve() {
 		stdout.Printf("You can now upload files to: http://%s%s", address, basepath)
 	}
 
-	if err = http.Serve(listener, nil); err != nil {
-		stderr.Fatalf("Unable to serve: %s", err)
+	var g run.Group
+
+	g.Add(func() error {
+		if err = s.Serve(listener); err != nil {
+			stderr.Fatalf("Unable to serve: %s", err)
+			return err
+		}
+		return nil
+	}, func(error) {
+		// TODO(rbastic): externalize shutdown timeout? for now just 30 mins? i don't know.
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*600)
+		defer cancel()
+		stderr.Fatalf("httpserver shutting down %s", s.Shutdown(ctx))
```

Comment:
Using Fatalf seems a bit harsh here since it forces a process to exit using os.Exit if I recall correctly. Can we use less aggressive logging statement here?

Reply:
Small followup: changed to stderr.Printf and chose 'shutdown-timeout' for the CLI parameter name.